raft/fuzz_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+import (
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+	pb "go.etcd.io/etcd/raft/v3/raftpb"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var initter sync.Once
+
+func initFunc() {
+	SetLogger(discardLogger)
+}
+
+func getMsgType(i int) pb.MessageType {
+	allTypes := map[int]pb.MessageType{0: pb.MsgHup,
+		1:  pb.MsgBeat,
+		2:  pb.MsgProp,
+		3:  pb.MsgApp,
+		4:  pb.MsgAppResp,
+		5:  pb.MsgVote,
+		6:  pb.MsgVoteResp,
+		7:  pb.MsgSnap,
+		8:  pb.MsgHeartbeat,
+		9:  pb.MsgHeartbeatResp,
+		10: pb.MsgUnreachable,
+		11: pb.MsgSnapStatus,
+		12: pb.MsgCheckQuorum,
+		13: pb.MsgTransferLeader,
+		14: pb.MsgTimeoutNow,
+		15: pb.MsgReadIndex,
+		16: pb.MsgReadIndexResp,
+		17: pb.MsgPreVote,
+		18: pb.MsgPreVoteResp}
+	return allTypes[i%len(allTypes)]
+}
+
+// All cases in shouldReport represent known errors in etcd
+// as these are reported via manually added panics.
+func shouldReport(err string) bool {
+	if strings.Contains(err, "stepped empty MsgProp") {
+		return false
+	}
+	if strings.Contains(err, "Was the raft log corrupted, truncated, or lost?") {
+		return false
+	}
+	if strings.Contains(err, "ConfStates not equivalent after sorting:") {
+		return false
+	}
+	if strings.Contains(err, "term should be set when sending ") {
+		return false
+	}
+	if (strings.Contains(err, "unable to restore config")) && (strings.Contains(err, "removed all voters")) {
+		return false
+	}
+	if strings.Contains(err, "ENCOUNTERED A PANIC OR FATAL") {
+		return false
+	}
+	if strings.Contains(err, "index, ") && strings.Contains(err, ", is out of range [") {
+		return false
+	}
+	// This string is found in raft.go because we change all
+	// occurrences from panic(err) to panic("GOT A FUZZ ERROR").
+	// This is done in build.sh as a simple solution to catch
+	// all the panic(err)'s that would otherwise be fuzz
+	// blockers.
+	if strings.Contains(err, "GOT A FUZZ ERROR") {
+		return false
+	}
+
+	return true
+}
+
+func catchPanics() {
+	if r := recover(); r != nil {
+		var err string
```

Comment:
Usually `err` is error type. 
```suggestion
		var errMsg string
```